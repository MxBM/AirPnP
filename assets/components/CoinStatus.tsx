import React from 'react';
import { View, TouchableOpacity, StyleSheet, Text } from 'react-native';
import colors from '../../theme/colors';
import Coin from '../images/svg/Coin.svg';
import Add from '../images/svg/Add.svg';

/*TODO: Create one unified status component for Health, Currency and Energy, and just pass in props for
icon? We'll also need a way to conditional have the logic for each respective status, maybe by having a string as a prop
that will allow us to state what type of function we'll need for the component. 
*/

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  capsuleContainer: {
    flexDirection: 'row',
    position: 'relative',
    alignContent: 'center',
    justifyContent: 'center',
    borderWidth: 4,
    borderColor: colors.black,
    backgroundColor: colors.setStatus,
    borderRadius: 40,
    padding: 8
  },
  icon: {
    height: 35, //Not the best way to do this.
    width: 30
  },
  textContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingLeft: 5,
    paddingRight: 5
  }
});

/*
TODO: Integrate coin system, I beleive that basic counter functions will suffice integrated with the backend (DB). Simple functions
to add coins and maybe even implementing a way for user's to buy coins with real money (NOT VIP).
*/

export default function CoinStatus() {
  return (
    <View style={styles.container}>
      <View style={styles.capsuleContainer}>
        <Coin style={styles.icon} />
        <View style={styles.textContainer}>
          <Text style={{ fontWeight: '900', fontSize: 22 }}>200</Text>
        </View>
        <TouchableOpacity onPress={() => console.log('Coin Status')}>
          <Add style={styles.icon} />
        </TouchableOpacity>
      </View>
    </View>
  );
}
