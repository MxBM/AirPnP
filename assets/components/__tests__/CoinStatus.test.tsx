import React from 'react';
import renderer from 'react-test-renderer';

import CoinStatus from '../CoinStatus';

it('renders correctly', () => {
  const tree = renderer.create(<CoinStatus />).toJSON();
  expect(tree).toMatchSnapshot();
});
