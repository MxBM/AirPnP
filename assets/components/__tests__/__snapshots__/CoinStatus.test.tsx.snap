// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#77767C",
        "borderColor": "#000000",
        "borderRadius": 40,
        "borderWidth": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
        "position": "relative",
      }
    }
  >
    <SvgMock
      style={
        Object {
          "height": 35,
          "width": 30,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "900",
          }
        }
      >
        200
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <SvgMock
        style={
          Object {
            "height": 35,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
